js/index8.js

```diff
@@ -548,15 +548,647 @@ console.log(uniqueSortedCourses); // ['biologie', 'informatică', 'limba englez
+}
+
+console.log(component);
+
+const componentIndex = componente.findIndex(
+  (componenta) => componenta.id === "asdf"
+);
+
+console.log(componentIndex);
+
+console.log("-------------------- every & some -------------------");
+
+console.log(filterArr.every((x) => x > -1000));
+
+console.log(filterArr.some((x) => x > 1000));
+
+console.log("------------------ reduce -----------------------");
+
+const result1 = filterArr.reduce((acc, value) => {
+  acc += value;
+  return acc;
+}, 0);
+
+console.log(result1);
+
+const words = [
+  "calculator",
+  "laptop",
+  "monitor",
+  "calculator",
+  "monitor",
+  "mouse",
+  "monitor",
+];
+
+console.log(`-metoda 1-"hasOwnProperty cu acc de la 0"(mai curat...!!!)`);
+
+const restul2 = words.reduce((acc, value) => {
+  if (!acc.hasOwnProperty(value)) {
+    acc[value] = 0;
+  }
+  acc[value]++;
+
+  return acc;
+}, {});
+
+console.log(restul2);
+
+console.log(`-metoda 2-"hasOwnProperty cu acc de la 1"`);
+
+const restul3 = words.reduce((acc, value) => {
+  if (!acc.hasOwnProperty(value)) {
+    acc[value] = 1;
+    return acc;
+  }
+  acc[value]++;
+  return acc;
+}, {});
+console.log(restul3);
+
+console.log(`-metoda 3-"=== undefined"`);
+
+const restul4 = words.reduce((acc, value) => {
+  const numbeOfRepetitions = acc[value];
+  if (numbeOfRepetitions === undefined) {
+    acc[value] = 0;
+  }
+  acc[value]++;
+  return acc;
+}, {});
+
+console.log(restul4);
+
+console.log(`-metoda 4-"!"`);
+
+const restul5 = words.reduce((acc, value) => {
+  const numbeOfRepetitions = acc[value];
+  if (!numbeOfRepetitions) {
+    acc[value] = 0;
+  }
+  acc[value]++;
+  return acc;
+}, {});
+
+console.log(restul5);
+
+console.log("------------------ sort ----------------------");
+console.log(`-!!! Atentie...!!! Modifica array-ul Original...!!!-`);
+
+const sortArr = [-51, 3, 27, 21, -68, 42, -37, 27];
+
+const arrCopy = [...sortArr];
+console.log(arrCopy.sort((a, b) => a - b));
+
+console.log(sortArr);
 
 // ------     Exercitii tema optional     ------
 console.log(`---------        Exercitii tema optional `);
 
 // - exercitiul-1 -
 console.log(`- exercitiul-1 -`);
 
+/*
+Implement the following methods using a for loop:
+ - map
+ - filter
+ - find
+ - every
+ - some
+*/
+
+console.log(`Implement the following methods using a for loop:
+ - map`);
+
+const radu = [0, 3, 23, 48, 78, 8, 9, 7];
+
+function getMap(array, callback) {
+  let newArray = [...array];
+  for (let i = 0; i < newArray.length; i++) {
+    newArray[i] = callback(array[i]);
+  }
+  return newArray;
+}
+const add = (value) => value + 2;
+
+console.log(radu);
+
+console.log(getMap(radu, add));
+
+console.log(`Implement the following methods using a for loop:
+ - filter`);
+
+function getFilter(array, callback) {
+  let newArray = [];
+  for (let arr of array) {
+    if (callback(arr) === true) {
+      newArray.push(arr);
+    } else if (callback(arr) === false) {
+      continue;
+    }
+  }
+  return newArray;
+}
```

Comment:
looks good, but there are some improvements that you can make:
1) since we expect the callback to return a boolean, there's no need to check === true
2) the else if is redundant

Reply:
ternar operator maybe ? 
(callback(arr))?newArray.push(arr):continue;
  return newArray;

or
  if (callback(arr)) {
      newArray.push(arr);
    }
      continue;
      return newArray;